src/games/strategy/engine/framework/startup/mc/ClientModel.java

```diff
@@ -335,35 +335,38 @@ private void startGameInNewThread(final byte[] gameData, final Map<String, INode
     final Set<IGamePlayer> playerSet = data.getGameLoader().createPlayers(playerMapping);
     final Messengers messengers = new Messengers(m_messenger, m_remoteMessenger, m_channelMessenger);
     m_game = new ClientGame(data, playerSet, players, messengers);
-    final Thread t = new Thread("Client Game Launcher") {
-      @Override
-      public void run() {
-        SwingUtilities.invokeLater(() -> JOptionPane.getFrameForComponent(m_ui).setVisible(false));
-        try {
-          // game will be null if we loose the connection
-          if (m_game != null) {
-            try {
-              data.getGameLoader().startGame(m_game, playerSet, false);
-              data.testLocksOnRead();
-            } catch (final Exception e) {
-              ClientLogger.logQuietly(e);
+    new Thread(() -> {
+      SwingUtilities.invokeLater(() -> JOptionPane.getFrameForComponent(m_ui).setVisible(false));
+      try {
+        // game will be null if we loose the connection
+        if (m_game != null) {
+          try {
+            data.getGameLoader().startGame(m_game, playerSet, false, () -> {
               m_game.shutDown();
```

Comment:
1. It's a pretty blatant copy-paste to repeat these 4 lines. You could at least declare a `Runnable closeAction` instance, and call `closeAction.run()` in the exception handler to avoid the duplication.
2. If `startGame` swallows any exceptions thrown, and executes our exception action, why are we catching an exception here at all? It seems like the catch block here is dead code, which was the bug to begin with (that the shutdown code was no longer being executed, and will never be executed)